content/blog/2021/04/2021-04-20-tekton-plugin.adoc

```diff
@@ -0,0 +1,178 @@
+
+This means that you can version your pipelines in git. It also means you can benefit from the link:https://jenkins-x.io/v3/develop/pipelines/editing/#ide-support[various IDE plugins] available for Tekton such as link:https://github.com/redhat-developer/vscode-tekton#tekton-pipelines-extension--[VS Code] and link:https://plugins.jetbrains.com/plugin/14096-tekton-pipelines-by-red-hat[IDEA] so that you get auto completion, formatting and validation while editing the YAML.
+
+So you can use the usual Git provider support in Jenkins to clone the git repository that contains then Tekton YAML file then reference the file by name.
+
+
+== Reusing Pipelines from the Tekton Catalog
+
+The link:https://github.com/tektoncd/catalog[Tekton Catalog] defines a ton of Tekton Tasks you can reuse in your pipelines
+
+We have found when it comes to a microsevices style architecture you end up with lots of repositories and pipelines. Then using a link:https://jenkins-x.io/v3/devops/patterns/pipeline_as_code/[Pipeline As Code] pattern with link:https://jenkins-x.io/v3/devops/patterns/gitops/[GitOps] we want to link:https://jenkins-x.io/v3/devops/patterns/version_everything/[Version Everything] but also make it easy for any repository to use any version of any task or pipeline.
+
+e.g. you may have many repositories using the current version of a pipeline but want to try out a new change to the pipeline in just 1 repository to verify it works well; then if it does, incrementally roll that change out to more repositories.
+
+This can make it hard trying to reuse as much as you can across the different git repositories while also minimising the number of versions and forks of git repositories you have and simplifying the maintenance of all of the pipelines.
+
+We have found on the link:https://jenkins-x.io/[Jenkins X project] that link:https://jenkins-x.io/blog/2021/02/25/gitops-pipelines/[a nice way to do this via GitOps] such that we reference versioned Tekton Tasks and Pipelines in git so that they are easy to link:https://jenkins-x.io/v3/develop/pipelines/catalog/#referencing-a-task-or-step[reuse] or link:https://jenkins-x.io/v3/develop/pipelines/catalog/#overriding-a-pipeline-step-locally[override].
+
+So we reuse Tasks and Pipelines via the link:https://jenkins-x.io/v3/develop/pipelines/catalog/#sourceuri-notation[uses: image notation] which lets us keep all of our Tekton Tasks and Pipelines in vanilla Tekton YAML; so that the IDE completion and validation works - but we can easily reuse Tasks or steps from libraries while also link:https://jenkins-x.io/v3/devops/patterns/version_everything/[Versioning Everything]
+
+**Note** that if wish to reuse steps/tasks via the link:https://jenkins-x.io/v3/develop/pipelines/catalog/#sourceuri-notation[uses: image notation] then you must click the `Tekton Catalog` flag in your Job definition which will then resolve the `uses:` clause with the actual step/task.
+
+== What is Jenkins X?
```

Comment:
It might make sense to split to two blogpost, but no strong opinion

Reply:
I'd rather it just be 1 really as then I don't have to create another demo ;)